src/renderers/dom/ReactDOMStringRenderer.js

```diff
@@ -36,10 +40,13 @@ function renderToString(element) {
  * See https://facebook.github.io/react/docs/react-dom-server.html#rendertostaticmarkup
  */
 function renderToStaticMarkup(element) {
-  invariant(
-    React.isValidElement(element),
-    'renderToStaticMarkup(): You must pass a valid ReactElement.',
-  );
+  const disableNewFiberFeatures = ReactFeatureFlags.disableNewFiberFeatures;
+  if (disableNewFiberFeatures) {
+    invariant(
```

Comment:
This still seems too early. It only validates top-level element, but not its children. What I was referring to is that [we have this validation in PartialRenderer](https://github.com/gzhappysky/react/blob/02101d5d4acccce66b303535aeea5bb07d9a384f/src/renderers/shared/server/ReactPartialRenderer.js#L310) and it should probably throw in some cases otherwise. Validation should happen as we `resolve` children rather than at the top.